src/Clock.tsx

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
+
+type Props = {
+  clockName: string;
+};
+
+export const Clock:React.FC<Props> = ({ clockName }) => {
+  const [time, setTime] = useState(new Date());
+  const [prevClockName, setPrevClockName] = useState(clockName);
+
+  useEffect(() => {
+    const timerId = window.setInterval(() => {
+      const newTime = new Date();
+
+      console.info(newTime.toUTCString().slice(-12, -4));
+
+      setTime(newTime);
+    }, 1000);
+
+    return () => {
+      window.clearInterval(timerId);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (prevClockName !== clockName) {
+      console.debug(`Renamed to ${clockName}`);
+      setPrevClockName(clockName);
+    }
+  }, [clockName]);
```

Comment:
You don't need this check, this function will run if clockname from dependency array its changing, so you can remove the if statement and the cachedClockname as well.